src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java

```diff
@@ -51,13 +57,22 @@ public LZ4SerializationFactory(final ObjectSerializingFactory factory) {
         this.factory = factory;
     }
 
+    /**
+     * @return {@code false}, this serializer does not support reading back multiple objects from a
+     *         single stream where each object has been individually compressed, due to a limitation
+     *         in {@code lz-java}
+     */
+    public @Override boolean supportsStreaming() {
+        return false;
+    }
+
     @Override
     public RevObject read(ObjectId id, InputStream in) throws IOException {
         LZ4FastDecompressor decompressor = lz4factory.fastDecompressor();
         Checksum checksum = newChecksum();
-        try (LZ4BlockInputStream cin = new LZ4BlockInputStream(in, decompressor, checksum)) {
-            return factory.read(id, cin);
-        }
+        LZ4BlockInputStream cin = new LZ4BlockInputStream(in, decompressor, checksum);
```

Comment:
These seem to be created but not closed - is this correct?

I'm not seeing anything in the factory.read() interface to say what's the expected behaviour.

Reply:
It's correct, the uncompressing stream decorator shall not close the argument stream